prov/sm2/src/sm2_atom.h

```diff
@@ -0,0 +1,277 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * ----------------[Copyright from inclusion of MPICH code]----------------
+ *
+ * The following is a notice of limited availability of the code, and disclaimer
+ * which must be included in the prologue of the code and in all source listings
+ * of the code.
+ *
+ * Copyright Notice
+ *  + 2002 University of Chicago
+ *
+ * Permission is hereby granted to use, reproduce, prepare derivative works, and
+ * to redistribute to others.  This software was authored by:
+ *
+ * Mathematics and Computer Science Division
+ * Argonne National Laboratory, Argonne IL 60439
+ *
+ * (and)
+ *
+ * Department of Computer Science
+ * University of Illinois at Urbana-Champaign
+ *
+ *
+ * 			      GOVERNMENT LICENSE
+ *
+ * Portions of this material resulted from work developed under a U.S.
+ * Government Contract and are subject to the following license: the Government
+ * is granted for itself and others acting on its behalf a paid-up,
+ * nonexclusive, irrevocable worldwide license in this computer software to
+ * reproduce, prepare derivative works, and perform publicly and display
+ * publicly.
+ *
+ * 				  DISCLAIMER
+ *
+ * This computer code material was prepared, in part, as an account of work
+ * sponsored by an agency of the United States Government.  Neither the United
+ * States, nor the University of Chicago, nor any of their employees, makes any
+ * warranty express or implied, or assumes any legal liability or responsibility
+ * for the accuracy, completeness, or usefulness of any information, apparatus,
+ * product, or process disclosed, or represents that its use would not infringe
+ * privately owned rights.
+ *
+ * This file provides shims between the opal atomics interface and the C11
+ * atomics interface. It is intended as the first step in moving to using C11
+ * atomics across the entire codebase. Once all officially supported compilers
+ * offer C11 atomic (GCC 4.9.0+, icc 2018+, pgi, xlc, etc) then this shim will
+ * go away and the codebase will be updated to use C11's atomic support
+ * directly. This shim contains some functions already present in atomic_impl.h
+ * because we do not include atomic_impl.h when using C11 atomics. It would
+ * require alot of #ifdefs to avoid duplicate definitions to be worthwhile.
```

Comment:
The paragraph after the disclaimer doesn't apply to the code and can be removed.  This is just a comment, not the disclaimer itself.

Reply:
Removed Disclaimer, and the comment.

Reply:
No - keep the disclaimer.  That's called out as required higher up in the license.  Only remove the comment, which doesn't apply to this code.